src/devices/src/virtio/net/device.rs

```diff
@@ -80,6 +85,28 @@ fn init_vnet_hdr(buf: &mut [u8]) {
     }
 }
 
+fn generate_mac_address() -> Option<MacAddr> {
+    let mut random_mac: [u8; MAC_ADDR_LEN] = [0; MAC_ADDR_LEN];
+    let (left, right) = random_mac.split_at_mut(3);
+    left.copy_from_slice(&KVM_OUI);
+    let file_result = File::open(Path::new(RANDOM_GENERATOR));
```

Comment:
What about using the rand implementations from utils::rand to generate the needed random stuff?

We would just need to do
```
right.copy_from_slice(rand::rand_alphanumerics(3).as_bytes());
```
You would get rid of all the file manipulations.

Reply:
I saw this one too while developing, the only issue is that it generates chars only between 0-9/a-z/A-Z which only covers a subset of 2^8 possible values per byte.
On the other hand since we are generating 3 bytes 62x62x62 options should be enough options for network devices.